tests/container/test_heap.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import absolute_import
+import random
+import sys
+import pytest
+import six
+from tchannel.container import heap
+
+from tchannel.container.heap import HeapOperation
+
+
+class IntHeap(HeapOperation):
+    def __init__(self):
+        self.values = []
+
+    def size(self):
+        return len(self.values)
+
+    def lt(self, i, j):
+        return self.values[i] < self.values[j]
+
+    def push(self, x):
+        self.values.append(x)
+
+    def pop(self):
+        n = len(self.values)
+        item = self.values[n - 1]
+        del self.values[n - 1]
+        return item
+
+    def swap(self, i, j):
+        self.values[i], self.values[j] = self.values[j], self.values[i]
+
+
+n = 1000
+
+
+@pytest.fixture
+def int_heap():
+    return IntHeap()
+
+
+def create_values(n):
+    values = []
+    for i in six.moves.range(n):
+        values.append(i)
+
+    random.shuffle(values)
```

Comment:
Let's make this slightly more than "all numbers from 1 to N, each occurring exactly once".

I would create a fixture that represents the list of numbers for that test run. Parameterize it so that you try the tests against different collections of numbers. Try to have at least the following

- What you have right now. Just 1000 numbers with no duplication.
- 100 random numbers between 1 and 1000 (so that there are many duplicates).
- Numbers from opposite ends -- some very small numbers, some very large numbers.

Then in the tests where you're doing `create_values(n)`, just use that fixture instead. The test will be run multiple times with the different lists.

See http://pytest.org/latest/example/parametrize.html#indirect-parametrization-with-multiple-fixtures.